src/connectors/toggle-refinement/connectToggleRefinement.js

```diff
@@ -256,64 +275,34 @@ export default function connectToggleRefinement(renderFn, unmountFn = noop) {
                 )
                 .filter(v => v !== undefined)
             : [];
-          const offFacetValue = {
+          offFacetValue = {
             isRefined:
               offData.length > 0 ? offData.every(v => v.isRefined) : false,
             count:
               offData.length === 0
                 ? allFacetValues.reduce((total, { count }) => total + count, 0)
                 : offData.reduce((acc, v) => acc + v.count, 0),
           };
-
-          // what will we show by default,
-          // if checkbox is not checked, show: [ ] free shipping (countWhenChecked)
-          // if checkbox is checked, show: [x] free shipping (countWhenNotChecked)
-          const nextRefinement = isRefined ? offFacetValue : onFacetValue;
-
-          value = {
-            name: attribute,
-            isRefined,
-            count: nextRefinement === undefined ? null : nextRefinement.count,
-            onFacetValue,
-            offFacetValue,
-          };
-        } else {
-          const isRefined =
-            on && on.every(v => state.isDisjunctiveFacetRefined(attribute, v));
-
-          // no need to refine anything at init if no custom off values
-          if (hasAnOffValue) {
-            // Add filtering on the 'off' value if set
-            if (!isRefined) {
-              const currentPage = helper.state.page;
-              if (off) {
-                off.forEach(v =>
-                  helper.addDisjunctiveFacetRefinement(attribute, v)
-                );
-              }
-
-              helper.setPage(currentPage);
-            }
+        } else if (hasAnOffValue && !isRefined) {
+          const currentPage = helper.state.page;
+          if (off) {
+            off.forEach(v =>
+              helper.addDisjunctiveFacetRefinement(attribute, v)
+            );
           }
 
-          const onFacetValue = {
-            isRefined,
-            count: 0,
-          };
+          helper.setPage(currentPage);
+        }
 
-          const offFacetValue = {
-            isRefined: hasAnOffValue && !isRefined,
-            count: 0,
-          };
+        const nextRefinement = isRefined ? offFacetValue : onFacetValue;
 
-          value = {
-            name: attribute,
-            isRefined,
-            count: null,
-            onFacetValue,
-            offFacetValue,
-          };
-        }
+        const value = {
+          name: attribute,
+          isRefined,
+          count: results && nextRefinement ? nextRefinement.count : null,
+          onFacetValue,
+          offFacetValue,
+        };
```

Comment:
@Haroenv I've removed duplicated code as much as I could and replaced it with some ternaries, let me know if it's still readable